torch/nn/parallel/distributed.py

```diff
@@ -552,11 +552,15 @@ def __init__(
         gradient_as_bucket_view=False,
         static_graph=False,
     ):
-
         super(DistributedDataParallel, self).__init__()
         Joinable.__init__(self)
         self.logger = None
-        if not any((p.requires_grad for p in module.parameters())):
+        if hasattr(module, "_ddp_params_and_buffers_to_ignore"):
```

Comment:
could you explain the reasoning for this change?

Reply:
this seems like an alternative to https://github.com/pytorch/pytorch/pull/81136, 
there are some sanitization logics (check device etc) that shouldn't apply to sharded tensors. However, even if we set that sharded tensor in params_and_buffers to ignore, sanitization is still run on them, 

this would bypass that

Reply:
we want to reuse the _module_parameters (where we ignore ignored ones). this is setting these as property of class to be reused